x-pack/plugins/fleet/server/services/epm/kibana/assets/install.ts

```diff
@@ -156,9 +166,67 @@ export async function installKibanaAssetsAndReferences({
     kibanaAssets,
   });
 
+  await tagKibanaAssets({
+    savedObjectTagAssignmentService,
+    savedObjectTagClient,
+    kibanaAssets,
+    pkgTitle,
+  });
+
   return installedKibanaAssetsRefs;
 }
 
+export async function tagKibanaAssets({
+  savedObjectTagAssignmentService,
+  savedObjectTagClient,
+  kibanaAssets,
+  pkgTitle,
+}: {
+  savedObjectTagAssignmentService: IAssignmentService;
+  savedObjectTagClient: ITagsClient;
+  kibanaAssets: Record<KibanaAssetType, ArchiveAsset[]>;
+  pkgTitle: string;
+}) {
+  const taggableAssets = Object.entries(kibanaAssets).flatMap(([assetType, assets]) => {
+    if (!taggableTypes.includes(assetType as KibanaAssetType)) {
+      return [];
+    }
+
+    if (!assets.length) {
+      return [];
+    }
+
+    return assets;
+  });
+
+  const managedTagName = 'Managed';
+  const tagColor = '#FFFFFF';
+  const allTags = await savedObjectTagClient.getAll();
+  let managedTag = allTags.find((tag) => tag.name === managedTagName);
+  if (!managedTag) {
+    managedTag = await savedObjectTagClient.create({
```

Comment:
I'd say potentially yes, especially if the concurrent calls are done from different Kibana instances?

ATM the tagClient does not accept specifying ids,

https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/saved_objects_tagging/server/services/tags/tags_client.ts#L27

But maybe it's necessary for your use case, to allow to specify the id + override:true to the underlying SO create call, just to avoid this risk?

Reply:
Yes, I think it would be a nice improvement to add. Should we raise an issue for this?